blockchain/validate.go

```diff
@@ -742,6 +742,14 @@ func (b *BlockChain) checkBlockHeaderContext(header *wire.BlockHeader, prevNode
 	// block.
 	blockHeight := prevNode.height + 1
 
+	// Ensure the header commits to the correct height based on the height it
+	// actually connects in the blockchain.
+	if int64(header.Height) != blockHeight {
+		errStr := fmt.Sprintf("Block header height invalid; expected %v"+
```

Comment:
might as well make the first character here uncapitalized to match the rest of the errors

Reply:
I'll go ahead and fix the error message.

Reply:
Improved the error too while changing it.